### PR TITLE
Updated lps25hb to handle i2c errors

### PR DESCRIPTION
**Pull request by Cristiana959**
### Pull Request Overview

This pull request adds i2c error verification for the lps25hb drivers.

### Testing Strategy

This pull request was not tested as I do not own the sensor.

### TODO or Help Wanted

N/A

### Documentation Updated

- [x]  No updates required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: capsules/src/lps25hb.rs [hunk 1]
```diff
@@ -149,20 +155,39 @@ impl<'a> LPS25HB<'a> {
...
+                self.state.set(State::TakeMeasurementInit);
+                Ok(())
+            }
+        })
     }
 }
 
 impl i2c::I2CClient for LPS25HB<'_> {
     fn command_complete(&self, buffer: &'static mut [u8], _status: Result<(), i2c::Error>) {
+        if _status != Ok(()) {
```
Comment (bradjc) [inline capsules/src/lps25hb.rs:173]:
```suggestion
    fn command_complete(&self, buffer: &'static mut [u8], status: Result<(), i2c::Error>) {
        if status != Ok(()) {
```

File: capsules/src/lps25hb.rs [hunk 2]
```diff
@@ -171,27 +196,88 @@ impl i2c::I2CClient for LPS25HB<'_> {
...
-                let _ = self.i2c.write(buffer, 1);
-                self.state.set(State::TakeMeasurementClear);
+                if let Err((error, buffer)) = self.i2c.write(buffer, 1) {
+                    self.state.set(State::Idle);
+                    self.buffer.replace(buffer);
+                    self.owning_process.map(|pid| {
+                        let _ = self.apps.enter(*pid, |_app, upcalls| {
+                            upcalls
+                                .schedule_upcall(
+                                    kernel::into_statuscode(Err(error.into())),
```
Comment (bradjc) [inline capsules/src/lps25hb.rs:206]: The first argument is the index of the upcall. Since there is only one upcall, this should always be 0.

**Review by bradjc — CHANGES_REQUESTED** [review]

**Review by bradjc — APPROVED** [review]

**Review by lschuermann — APPROVED** [review]

**Comment by bradjc** [comment]
Need to update to `kernel::errorcode::into_statuscode`.

**Review by alistair23 — APPROVED** [review]

**Review by bradjc — APPROVED** [review]

**Pull request merged**